marekventur/dependency-updater
# Update "sinon" to version 1.17.5

<pre>1.17.5 / 2016-07-26
===================

  * Merge pull request [#1097](https://github.com/cjohansen/Sinon.JS/issues/1097) from ClintFMullins/add-assert-to-reduce-called-once-with-confusion
    Add assertion check for too many args with calledOnce/Twice/Thrice
  * add assertion check for too many args with calledOnce/Twice/Thrice
  * Merge pull request [#1092](https://github.com/cjohansen/Sinon.JS/issues/1092) from fatso83/use-details-tag
    Use <details> tag in issue template
  * Use <details> tag in issue template 📝
    [demo](https://twitter.com/ericclemmons/status/749223563790471169)
  * Merge pull request [#1087](https://github.com/cjohansen/Sinon.JS/issues/1087) from jonnyreeves/feature/cjsify-sinon-mock
    CJSify mock tests
  * Merge pull request [#1086](https://github.com/cjohansen/Sinon.JS/issues/1086) from jonnyreeves/feature/cjsify-match-tests
    CJSify match tests
  * CJSify mock tests
    Removes all `sinon.*` references from the test suite
  * CJSify match tests
    Removes all `sinon.*` references from the test suite
  * Merge pull request [#1085](https://github.com/cjohansen/Sinon.JS/issues/1085) from jonnyreeves/feature/cjsify-typeof-extend-tests
    Feature/cjsify typeof extend tests
  * CJSify typeOf tests
    Removes all `sinon.*` references from the test suite
  * CJSify extend tests
    Removes all `sinon.*` references from the test suite
  * Merge pull request [#1084](https://github.com/cjohansen/Sinon.JS/issues/1084) from jonnyreeves/feature/cjsify-collection-tests
    CJSify collection tests
  * CJSify collection tests
    Removes all `sinon.*` references from both the tests and the collection module itself.  A bunch of tests around stubbing and mocking had to be rewritten as it's no longer possible to monkeypatch `mock` and `stub` - the replacement tests are now based on the expected behavior.
  * Merge pull request [#1082](https://github.com/cjohansen/Sinon.JS/issues/1082) from fatso83/remove-nuget
    The NuGet repo has not been updated for two years.
    Furthermore, some lines I felt was uneeded was
    also removed or altered somewhat.
  * Improve release process ([#1083](https://github.com/cjohansen/Sinon.JS/issues/1083))
  * Remove NuGet reference
    The NuGet repo has not been updated for two years.
    Furthermore, some lines I felt was uneeded was
    also removed or altered somewhat.

2.0.0-pre.2 / 2016-07-07
========================

  * 2.0.0-pre.2
  * Update changelog and authors
  * CJSify sinon.call tests ([#1079](https://github.com/cjohansen/Sinon.JS/issues/1079))
    Removes all `sinon.*` references from the tests.
  * CJSify sinon.calledInOrder tests ([#1080](https://github.com/cjohansen/Sinon.JS/issues/1080))
    Removes all `sinon.*` references from the tests and moves the suite into `util/core` where is belongs.
  * CJSify get-config tests ([#1081](https://github.com/cjohansen/Sinon.JS/issues/1081))
    Removes all `sinon.*` references from the tests
  * CJSify sinon.assert tests ([#1078](https://github.com/cjohansen/Sinon.JS/issues/1078))
    Removes all `sinon.*` references from the tests.
  * Resolve test failure in node 0.10.x ([#1073](https://github.com/cjohansen/Sinon.JS/issues/1073))
    The test for "adds all object methods to fake array" for
    collection.stub failed intermittently because the order of methods returned
    from `walk` is not guaranteed.
  * Expose `sinon.assert` on sandbox instances. ([#1076](https://github.com/cjohansen/Sinon.JS/issues/1076))
    Fixes [#1075](https://github.com/cjohansen/Sinon.JS/issues/1075).
  * Merge pull request [#1074](https://github.com/cjohansen/Sinon.JS/issues/1074) from Gudahtt/add-resetbehavior-and-resethistory-to-sandbox-api
    Add resetBehavior and resetHistory to sandbox API
  * Add resetBehavior and resetHistory to sandbox API ([#1072](https://github.com/cjohansen/Sinon.JS/issues/1072))
  * Fix incorrect inline function names
  * Merge pull request [#1071](https://github.com/cjohansen/Sinon.JS/issues/1071) from lucasfcosta/fix-issue-283-tests
    Fix issue 283 tests
  * Fix calledOnce on tests for [#283](https://github.com/cjohansen/Sinon.JS/issues/283). This closes [#283](https://github.com/cjohansen/Sinon.JS/issues/283).
  * Merge pull request [#1068](https://github.com/cjohansen/Sinon.JS/issues/1068) from jackbrown/patch-1
    Add `sandbox.reset()` description
  * Add sandbox.reset() docs
  * Add a line recommending how to pronounce.
  * Merge pull request [#1069](https://github.com/cjohansen/Sinon.JS/issues/1069) from fatso83/vietnogi-loaded-fix
    Use event loaded instead of error event for HTTP error codes
  * Improve tests based on PR feedback
    Based on fearphage's feedback in 1031 an uneeded
    test was deleted and the clarity of another was improved.
  * Allow xhr.respond(0) to simulate a network failure and call onerror
  * Use event loaded instead of error event for code like 403, 500, etc.
    Simplify abort handling and fix progress results. Also update tests.
  * Merge pull request [#1060](https://github.com/cjohansen/Sinon.JS/issues/1060) from fearphage/fix-failing-tests
    Added timeouts to ensure tests pass
  * Fix invalid markdown in fake-timers.ms ([#1054](https://github.com/cjohansen/Sinon.JS/issues/1054))
  * Do not invoke getters in walk ([#1059](https://github.com/cjohansen/Sinon.JS/issues/1059))
    When stubbing a prototype, the `walk` implementation accesses getters to
    obtain the value of each property to pass it to the iterator function.
    This value is not needed by any of the iterator implementations anymore,
    but invoking the getter may have side effects or throw exceptions.
    With this change, values are no longer obtained from the objects. It
    keeps the code to pass a different target object to the iterator which
    allows to obtain the value in the iterator implementation if necessary.
  * ReactNative compatibility. Allow sinon fakeServer to run in React Native ([#1052](https://github.com/cjohansen/Sinon.JS/issues/1052))
    ReactNative compatibility. Allow sign fakeServer to run in React Native environment either on device or browser debug.
    ReactNative has xhr support via polyfill. But the check for supporting xhr looked and withEntitlements which is failing in RN.
    ReactNative does not have window.location so a default assignment and check for undefined window.location was needed.
  * added timeouts to ensure tests pass
  * Run tests on stable Node 6 instead of unstable Node 5
  * added tests to ensure only expected events are fired ([#1043](https://github.com/cjohansen/Sinon.JS/issues/1043))
  * Fixed formatting of issue template
  * Added note on using latest version
  * Fix onerror event triggering for fake xhr requests ([#1041](https://github.com/cjohansen/Sinon.JS/issues/1041))
    * added missing test coverage
    * trigger onerror event when available</pre>
